File: app/src/main/rust/message/what.rs
pub const INITIATE: i32 = 0;
pub const COUNTER_VALUE: i32 = 1;
pub const INCREASE_COUNTER: i32 = 2;
pub const DECREASE_COUNTER: i32 = 3;
pub const ERROR: i32 = 4;
pub const SNACK_BAR_MSG: i32 = 5;

pub fn to_string(what: i32) -> &'static str {
    match what {
        INITIATE => "INITIATE",
        COUNTER_VALUE => "COUNTER_VALUE",
        INCREASE_COUNTER => "INCREASE_COUNTER",
        DECREASE_COUNTER => "DECREASE_COUNTER",
        ERROR => "ERROR",
        SNACK_BAR_MSG => "SNACK_BAR_MSG",
        _ => "<UNKNOWN>",
    }
}

File: app/src/main/java/io/rousan/globalcounter/message/What.java
package io.rousan.globalcounter.message;

public final class What {
    public final static int INITIATE = 0;
    public final static int COUNTER_VALUE = 1;
    public final static int INCREASE_COUNTER = 2;
    public final static int DECREASE_COUNTER = 3;
    public final static int ERROR = 4;
    public final static int SNACK_BAR_MSG = 5;

    public static String toString(int what) {
        switch (what) {
            case INITIATE: {
                return "INITIATE";
            }
            case COUNTER_VALUE: {
                return "COUNTER_VALUE";
            }
            case INCREASE_COUNTER: {
                return "INCREASE_COUNTER";
            }
            case DECREASE_COUNTER: {
                return "DECREASE_COUNTER";
            }
            case ERROR: {
                return "ERROR";
            }
            case SNACK_BAR_MSG: {
                return "SNACK_BAR_MSG";
            }
            default: {
                return "<UNKNOWN>";
            }
        }
    }
}